src/nvim/if_cscope.c

```diff
@@ -1772,11 +1777,14 @@ static int cs_read_prompt(int i)
       if (bufpos < maxlen - 1 && vim_isprintc(ch)) {
         // lazy buffer allocation
         if (buf == NULL) {
-          buf = xmalloc(maxlen);
+          assert(maxlen >= 0);
+          buf = xmalloc((size_t)maxlen);
         }
         {
           /* append character to the message */
-          buf[bufpos++] = ch;
+          assert(ch >= 0);
+          assert(ch <= CHAR_MAX);
```

Comment:
To me, one single assert is more conceptually clear, and reads better.
There are complicated asserts with multiple clauses. Augmenting code pollution by multiplying lines doesn't look like a good idea to me.
If you get a bug report for a failing assert, it's easy enough to split it then if you want, while debugging.
Also, note that many of the asserts will be simplified when dedicated macros to check for bounds/overflow are introduced.